packages/lab/src/__tests__/__e2e__/carousel/Carousel.cy.tsx

```diff
@@ -0,0 +1,120 @@
+import * as carouselStories from "@stories/carousel/carousel.stories";
+import { composeStories } from "@storybook/react";
+import { checkAccessibility } from "../../../../../../cypress/tests/checkAccessibility";
+
+const composedStories = composeStories(carouselStories);
+const { Default, WithActions, Controlled } = composedStories;
+describe("GIVEN a 100% width slides carousel", () => {
+  checkAccessibility(composedStories);
+  describe("WHEN the default is rendered with slides", () => {
+    it("SHOULD render carousel", () => {
+      cy.mount(<Default />);
+      cy.findByRole("region").should("exist");
+    });
+  });
+  describe("WHEN moving slides with buttons", () => {
+    it("SHOULD move to the next slide with button", () => {
+      cy.mount(<Default />);
+      cy.findAllByText("1 of 4").should("exist");
+      cy.findAllByRole("button", { name: "Previous slide" }).should(
+        "have.attr",
+        "aria-disabled",
+        "true",
+      );
+      cy.findAllByRole("button", { name: "Next slide" }).click();
+      cy.findAllByText("2 of 4").should("exist");
+    });
+
+    it("SHOULD move to the previous slide with button", () => {
+      cy.mount(<Default />);
+      // move one to the left
+      cy.findAllByRole("button", { name: "Next slide" }).click();
```

Comment:
note.. these buttons could do nothing and you would never know the test was failing.
See lines 17 and 18... it tests the intial state and on line 34 the same state,
We don't check for any in-between state.